### PR TITLE
make Miri work in rustc bootstrap stage 0

### PR DESCRIPTION
**Pull request by RalfJung**
Fixes https://github.com/rust-lang/rust/issues/52856

### CODE REVIEW

**Comment by RalfJung** [comment]
Ah... this breaks outside of the rustbuild environment because "miri-as-rustc" uses the wrong sysroot dir... namely something relative to where the binary lives.

**Comment by RalfJung** [comment]
This now works locally, and should also still work during bootstrap.

r? @oli-obk

File: README.md [hunk 1]
```diff
@@ -222,6 +222,13 @@ Moreover, Miri recognizes some environment variables:
 * `MIRI_TEST_FLAGS` (recognized by the test suite) defines extra flags to be
   passed to Miri.
 
+The following environment variables are internal, but used to communicate between
+different Miri binaries, and as such worht documenting:
```
Comment (RalfJung) [inline README.md:226]:
```suggestion
different Miri binaries, and as such worth documenting:
```

File: src/bin/miri.rs [hunk 2]
```diff
@@ -123,7 +122,47 @@ fn compile_time_sysroot() -> Option<String> {
     })
 }
 
+/// Execute a compiler with the given CLI arguments and callbacks.
+fn run_compiler(mut args: Vec<String>, callbacks: &mut (dyn rustc_driver::Callbacks + Send)) {
```
Comment (oli-obk) [inline src/bin/miri.rs:126]: Oh that's a nice convenience function. I wonder if we can upstream the non-sysroot part of it

Comment (RalfJung) [reply src/bin/miri.rs:126]: Yeah I wondered the same actually.
Any proposals for a name, given that `rustc_driver::run_compiler` is already taken?

**Pull request merged**